### DISCUSSION TITLE
Validation to manage users

### DISCUSSION THREAD

**Discussion by pippuccio76** [codeigniter4/shield#1028] [Q&A]
Hi , how can i set a validation rule to insert user with unique email ?

**Comment by datamweb** [comment]
Shield has applied these rules in the registration form by default.
```php
    'email' => [
        'label' => 'Auth.email',
        'rules' => [
            'required',
            'max_length[254]',
            'valid_email',
            'is_unique[auth_identities.secret]',
        ],
    ],
```
More info see https://shield.codeigniter.com/customization/validation_rules/#registration

But if you want to do this custom in a new controller, you can use the [RegisterController.php](https://github.com/codeigniter4/shield/blob/6bb80f7b2cc9f8efe58b49396127ebf75cd6aa8b/src/Controllers/RegisterController.php#L96-L102). To read more, refer to [CI4 Validation](https://codeigniter.com/user_guide/libraries/validation.html).
